src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/AppResourcesSI.cs

```diff
@@ -85,6 +81,29 @@ public byte[] GetText(string org, string app, string textResource)
             return ReadFileContentsFromLegalPath(_settings.AppBasePath + _settings.ConfigurationFolder + _settings.TextFolder, textResource);
         }
 
+        /// <inheritdoc />
+        public TextResource GetTexts(string org, string app, string language)
+        {
+            string pathTextsFolder = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.TextFolder;
+            byte[] textsAsByteArray = ReadFileContentsFromLegalPath(pathTextsFolder, $"resource.{language}.json");
```

Comment:
This method uses File.ReadAllBytes under the hood. I ran into a problem on files with Unicode ByteOrderMark (BOM) included when reading raw bytes since the first 3 bytes then where BOM bytes. This caused the Json.Net serializer to fail. I see you are using System.Text.Json.JsonSerializer perhaps this one is more resilient with regards to BOM?

Reply:
I can't say I know. I will do a test of the new code where I try to change encoding.

Reply:
I can confirm that it fails. Not sure how important it'll be to support UTF8 with BOM.

Reply:
I've changed the logic to read the file stream directly instead of converting from byte[] to string before deserializing. The built in logic can handle the BOM for us.